airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -398,4 +329,141 @@ private void continueAsNew(final ConnectionUpdaterInput connectionUpdaterInput)
+   */
+  private Duration getTimeToWait(UUID connectionId) {
+    // Scheduling
+    final ScheduleRetrieverInput scheduleRetrieverInput = new ScheduleRetrieverInput(connectionId);
+
+    final ScheduleRetrieverOutput scheduleRetrieverOutput = configFetchActivity.getTimeToWait(scheduleRetrieverInput);
+
+    return scheduleRetrieverOutput.getTimeToWait();
+  }
+
+  /**
+   * Check if a sync need to be run based on the workflow state.
+   */
+  private boolean needToProcessSync() {
+    return !workflowState.isUpdated() && !workflowState.isDeleted();
+  }
+
+  /**
+   * Creates a new job if it is not present in the input. If the jobId is specified in the input of
+   * the connectionManagerWorkflow, we will return it. Otherwise we will create a job and return its
+   * id.
+   */
+  private Long getOrCreateJobId(final ConnectionUpdaterInput connectionUpdaterInput) {
+    return Optional.ofNullable(connectionUpdaterInput.getJobId()).or(() -> {
+      final JobCreationOutput jobCreationOutput =
+          runMandatoryActivityWithOutput(
+              (input) -> jobCreationAndStatusUpdateActivity.createNewJob(input),
+              new JobCreationInput(
+                  connectionUpdaterInput.getConnectionId(), workflowState.isResetConnection()));
+      connectionUpdaterInput.setJobId(jobCreationOutput.getJobId());
+      return Optional.ofNullable(jobCreationOutput.getJobId());
+    }).get();
+  }
+
+  /**
+   * Create a new attempt for a given jobId
+   */
+  private Integer createAttemptId(long jobId) {
+    final AttemptCreationOutput attemptCreationOutput =
+        runMandatoryActivityWithOutput(
+            (input) -> jobCreationAndStatusUpdateActivity.createNewAttempt(input),
+            new AttemptCreationInput(
+                jobId));
+
+    return attemptCreationOutput.getAttemptId();
+  }
+
+  /**
+   * Generate the input that is needed by the job. It will generate the configuration needed by the
+   * job and will generate a different output if the job is a sync or a reset.
+   */
+  private GeneratedJobInput getJobInput() {
+    final SyncInput getSyncInputActivitySyncInput = new SyncInput(
+        attemptId,
+        jobId,
+        workflowState.isResetConnection());
+
+    final GeneratedJobInput syncWorkflowInputs = runMandatoryActivityWithOutput(
+        (input) -> getSyncInputActivity.getSyncWorkflowInput(input),
+        getSyncInputActivitySyncInput);
+
+    return syncWorkflowInputs;
+  }
+
+  /**
+   * Report the job as started in the job tracker and set it as running in the workflow internal
+   * state.
+   */
+  private void reportJobStarting() {
+    runMandatoryActivity(
+        (input) -> jobCreationAndStatusUpdateActivity.reportJobStart(input),
+        new ReportJobStartInput(
+            jobId));
+
+    workflowState.setRunning(true);
+  }
+
+  /**
+   * Start the child SyncWorkflow
+   */
+  private StandardSyncOutput runChildWorkflow(GeneratedJobInput jobInputs) {
```

Comment:
can we document why we use a child workflow in the docstring? lmk if this is documented somewhere and I've missed it.

Reply:
It is not documented a the moment. What do you mean by docstring? Is it a doc that should live in the docs folder of the OSS project or the cloud project or should it be in the comment? I took a note for the first option. I think that it would be too much explanation for a comment. The main idea of it was to avoid modifying the existing code for running the sync (the one that was launched by the scheduler) so having the same workflow as a child workflow made sense.

Reply:
what charles said!